File: Add Code Here/JAVA/LeetCode/283. Move Zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int temp=0, left=0, right=0;
        if(nums.length==0 || nums.length==1) return;
        for(int i=0; i<nums.length; ++i){

        if(nums[right]==0){
            ++right;
        }
            
        else {
            temp=nums[left];
            nums[left]=nums[right];
            nums[right]=temp;
            ++right;
            ++left;
        }
    }
}}

File: Add Code Here/CPP/283_Move_Zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int count=0,d=nums.size(),j=0;
        for(int i=0;i<d;i++){
            if(nums[i]==0){
                count++;
            }
            else{
                nums[j++]=nums[i];
            }
        }
        int i=1;
        while(count--){
            nums[d-i]=0;
            i++;
        }
    }
};
